lib/elastic_record/as_document.rb

```diff
@@ -4,48 +4,42 @@ def as_search_document(mapping_properties = elastic_index.mapping[:properties])
       mapping_properties.each_with_object({}) do |(field, mapping), result|
         value = value_for_elastic_search field, mapping, mapping_properties
 
-        if !value.nil? || (respond_to?(:provided?) && provided?(field))
+        unless value.nil?
           result[field] = value
         end
       end
     end
 
-    def as_partial_update_document
-      mapping_properties = elastic_index.mapping[:properties]
+    def as_partial_update_document(mapping_properties = elastic_index.mapping[:properties])
       changed_fields = respond_to?(:saved_changes) ? saved_changes.keys : changed
 
       changed_fields.each_with_object({}) do |field, result|
         if field_mapping = mapping_properties[field]
-          result[field] = value_for_elastic_search field, field_mapping, mapping_properties
+          result[field] = value_for_elastic_search field, field_mapping, mapping_properties, true
         end
       end
     end
 
-    def value_for_elastic_search(field, mapping, mapping_properties)
-      value = try field
-      return if value.nil?
-
-      value =
-        case mapping[:type]&.to_sym
-        when :object
-          object_mapping_properties = mapping_properties.dig(field, :properties)
-          value_for_elastic_search_object(value, object_mapping_properties)
-        when :nested
-          object_mapping_properties = mapping_properties.dig(field, :properties)
-          value.map { |entry| value_for_elastic_search_object(entry, object_mapping_properties) }
-        when :integer_range, :float_range, :long_range, :double_range, :date_range
-          value_for_elastic_search_range(value)
-        else
-          value
-        end
-
-      if value.present? || value == false
-        value
+    def value_for_elastic_search(field, mapping, mapping_properties, is_update = false)
+      return unless (value = try(field))
```

Comment:
Gah!  It took me too long to figure out this line was the problem.

Reply:
Sorry about that, I was moving the guards around a lot at first, and forgot to put this one back to the original nil check (vs falsey).